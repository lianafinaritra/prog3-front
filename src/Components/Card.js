
export default function Card( props ) {
    const { handleUpdate, item } = props;

    return(
        <div className="card"  data-testid="cardItem">
            <p>{item.text}</p>
            <div className="checkbox">
                {item.status === "TODO" && (
                    <div data-testid="checkbox">
                        <input type="checkbox" onClick={() => handleUpdate(item)} />
                    </div>
                )}
            </div>
        </div>
    )
}