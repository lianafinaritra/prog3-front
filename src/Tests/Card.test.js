import { render } from "@testing-library/react";
import Card from "../Components/Card";
import React from "react";
import  renderer  from "react-test-renderer";

let todoItem = {
  id: 1,
  text : "description",
  status : "TODO",
}

let doneItem = {
  id: 2,
  text: "description",
  status: "DONE",
};

describe('Card test', () => {
    const handleUpdate = jest.fn();

    it('show description', () => {
        const{ getByText } = render(
            <Card item={todoItem} handleUpdate={handleUpdate} key={todoItem.id}/>
        )
        expect(getByText("description")).toBeInTheDocument();
    });

    it('display checkbox when task is in TODO status', () => {
        const { getByTestId } = render(
            <Card item={todoItem} handleUpdate={handleUpdate} key={todoItem.id}/>
        );
        expect(getByTestId("checkbox")).toBeTruthy();
    });

    it("don't display checkbox when task is DONE ", () => {
        const { queryByTestId } = render(
            <Card item={doneItem} handleUpdate={handleUpdate} key={todoItem.id}/>
        );
        expect(queryByTestId("checkbox")).toBeNull();
    });

    it("should renders correctly", ()=>{
        const cardSnap = renderer.create(
            <Card item={doneItem} handleUpdate={handleUpdate} key={todoItem.id}/>
        )
        const card = cardSnap.toJSON();

        expect(card).toMatchSnapshot();
    })
});
  
