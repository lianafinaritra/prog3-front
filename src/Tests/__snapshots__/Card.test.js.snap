// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test should renders correctly 1`] = `
<div
  className="card"
  data-testid="cardItem"
>
  <p>
    description
  </p>
  <div
    className="checkbox"
  />
</div>
`;
